base/reflection.jl

```diff
@@ -460,6 +460,23 @@ true
 """
 ismutable(@nospecialize(x)) = (@_pure_meta; typeof(x).mutable)
 
+
+"""
+    ismutabletype(T) -> Bool
+
+Determine whether type `T` was declared as a struct type
+(i.e. using `mutable struct` keyword).
+
+!!! compat "Julia 1.7"
+    This function requires at least Julia 1.7.
+"""
+function ismutabletype(@nospecialize(t::Type))
+    t = unwrap_unionall(t)
```

Comment:
The similar `isstructtype` and `isprimitivetype` definitions below seem to be at-pure; should this definition be as well? :)

Reply:
The compiler should be able to prove that itself.